featurebyte/service/tile_manager.py

```diff
@@ -110,32 +112,44 @@ async def _progress_callback() -> None:
     async def _generate_tiles_on_demand_for_tile_spec(
         self,
         session: BaseSession,
-        tile_spec: TileSpec,
-        entity_table: str,
+        on_demand_tile_spec: OnDemandTileSpec,
         progress_callback: Optional[Callable[[], Coroutine[Any, Any, None]]] = None,
     ) -> None:
         tic = time.time()
         session = await session.clone_if_not_threadsafe()
         await self.generate_tiles(
             session=session,
-            tile_spec=tile_spec,
+            tile_spec=on_demand_tile_spec.tile_spec,
             tile_type=TileType.OFFLINE,
             start_ts_str=None,
             end_ts_str=None,
         )
         logger.debug(
             "Done generating tiles",
-            extra={"tile_id": tile_spec.tile_id, "duration": time.time() - tic},
+            extra={"tile_id": on_demand_tile_spec.tile_spec.tile_id, "duration": time.time() - tic},
         )
 
         tic = time.time()
-        await self.update_tile_entity_tracker(
-            session=session, tile_spec=tile_spec, temp_entity_table=entity_table
-        )
-        logger.debug(
-            "Done update_tile_entity_tracker",
-            extra={"tile_id": tile_spec.tile_id, "duration": time.time() - tic},
-        )
+        if on_demand_tile_spec.tracker_sql is not None:
+            await self.update_tile_entity_tracker(
+                session=session,
+                tile_spec=on_demand_tile_spec.tile_spec,
+                temp_entity_table=on_demand_tile_spec.tracker_sql,
+            )
+            logger.debug(
+                "Done update_tile_entity_tracker",
+                extra={
+                    "tile_id": on_demand_tile_spec.tile_spec.tile_id,
+                    "duration": time.time() - tic,
+                },
+            )
+        if on_demand_tile_spec.observation_table_id is not None:
+            await (
+                self.observation_table_tile_cache_service.add_aggregation_ids_for_observation_table(
```

Comment:
If the `aggregation_id` has not been computed for the observation table, we assume the tiles are not available and recompute. Otherwise, we skip the computation for the `aggregation_id`.